File: backend/app/schemas/review.py
# today_book/backend/app/schemas/review.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class ReviewCreate(BaseModel):
    type: str  # weekly/monthly/yearly
    period_start: str
    period_end: str
    title: Optional[str] = None
    content: Optional[str] = None
    accomplishments: Optional[list[str]] = None
    challenges: Optional[list[str]] = None
    next_plans: Optional[list[str]] = None
    mood_score: Optional[int] = None


class ReviewUpdate(BaseModel):
    title: Optional[str] = None
    content: Optional[str] = None
    accomplishments: Optional[list[str]] = None
    challenges: Optional[list[str]] = None
    next_plans: Optional[list[str]] = None
    mood_score: Optional[int] = None
    is_completed: Optional[bool] = None


class ReviewResponse(BaseModel):
    id: int
    user_id: int
    type: str
    period_start: str
    period_end: str
    title: Optional[str]
    content: Optional[str]
    accomplishments: Optional[list[str]]
    challenges: Optional[list[str]]
    next_plans: Optional[list[str]]
    mood_score: Optional[int]
    is_completed: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/api/review.ts
/* today_book/frontend/src/api/review.ts */
import apiClient from './client'

export interface Review {
  id: number
  type: string
  period_start: string
  period_end: string
  title?: string
  content?: string
  accomplishments?: string[] | string
  challenges?: string[] | string
  next_plans?: string[] | string
  mood_score?: number
  is_completed: boolean
  created_at: string
  updated_at: string
}

export const reviewApi = {
  list: (type?: string) => apiClient.get<Review[]>('/review/', type ? { params: { review_type: type } } : {}),
  get: (id: number) => apiClient.get<Review>(`/review/${id}`),
  create: (data: Partial<Review>) => apiClient.post<Review>('/review/', data),
  update: (id: number, data: Partial<Review>) => apiClient.put<Review>(`/review/${id}`, data),
  delete: (id: number) => apiClient.delete(`/review/${id}`),
  getTemplate: (type: string) => apiClient.get(`/review/template/${type}`),
  getStats: () => apiClient.get('/review/stats/summary'),
}
